influxql/ast.go

```diff
@@ -839,6 +839,46 @@ func (s *SelectStatement) HasWildcard() bool {
 	return false
 }
 
+// hasTimeDimensions returns whether or not the select statement has at least 1
+// where condition with time as the condition
+func (s *SelectStatement) hasTimeDimensions(node Node) bool {
+	switch n := node.(type) {
+	case *BinaryExpr:
+		if n.Op == AND || n.Op == OR {
+			return s.walkForTime(n.LHS) || s.walkForTime(n.RHS)
+		}
+		if ref, ok := n.LHS.(*VarRef); ok && strings.ToLower(ref.Val) == "time" {
+			return true
+		}
+		return false
+	case *ParenExpr:
+		// walk down the tree
+		return s.walkForTime(n.Expr)
+	default:
+		return false
+	}
+}
+
+// Validate checks certain edge conditions to determine if this is a valid select statment
+func (s *SelectStatement) Validate(tr targetRequirement) error {
```

Comment:
@otoolep I moved the if statements to their own function to clear up the parse function.  Bottom line is if we want these checks, it requires if statements.  I'm open to other suggestions if you feel the code is unbearable.